services/introspection/local.go

```diff
@@ -105,8 +106,18 @@ func (l *Local) Server(ctx context.Context, _ *ptypes.Empty, _ ...grpc.CallOptio
 	if err != nil {
 		return nil, errdefs.ToGRPC(err)
 	}
+	pid := os.Getpid()
```

Comment:
I think we don't unshare newpid for the short-live thread. It is safe to do that.

Reply:
Sorry, I don't get

Reply:
Sorry for my unclear comment.

I was thinking why not just hide the `pid` param in the `statPIDNS` function, 
since `pid` won't change in the entire life-cycle.

And I also consider that the result can be cached when plugin initializes.
But other random thought comes. Is it possible to change containerd's pidns like #7513? 
If the pidns changes, we can't cache it. :joy: